pkg/bpfassets/attacher/bpf_perf.go

```diff
@@ -57,8 +57,8 @@ var (
 // must be in sync with bpf program
 type ProcessBPFMetrics struct {
 	CGroupID       uint64
-	PID            uint64
-	TGID           uint64
+	ThreadPID      uint64 /* thread id */
+	PID            uint64 /* TGID of the threads, i.e. user space pid */
```

Comment:
looks like it's just naming convention change ? e.g TGID => PID (since it's user space)

Reply:
most of the work is done in ebpf, the userspace code is to use the tgid to track processes instead of threads.